conda/activate.py

```diff
@@ -9,15 +9,15 @@
 from tempfile import NamedTemporaryFile
 
 from .base.context import ROOT_ENV_NAME, context, locate_prefix_by_name
-context.__init__()  # oOn import, context does not include SEARCH_PATH. This line fixes that.
+context.__init__()  # On import, context does not include SEARCH_PATH. This line fixes that.
 
 try:
     from cytoolz.itertoolz import concatv, drop
 except ImportError:  # pragma: no cover
     from ._vendor.toolz.itertoolz import concatv, drop  # NOQA
 
 
-class Activator(object):
+class _Activator(object):
```

Comment:
We *could* make this `class _Activator(ABC)` and thus use `@abstractmethod` to automatically check for methods/properties being implemented.
This would add a little bit more boilerplate (which I would be fine with). But we would also need to look out for Python 2 intricacies (e.g., need to use `metaclass=ABCMeta`, the deprecated `@abstractproperty`, ...).
So, although I'd prefer using `ABC`, I'm also not eager to deal with PY2 stuff...

Reply:
Yeah that's why I didn't go in that direction, and just renamed `Activator` to `_Activator`.